pudl_scrapers/spiders/eia860m.py

```diff
@@ -0,0 +1,123 @@
+"""Spider for 860 M."""
+# -*- coding: utf-8 -*-
+from pathlib import Path
+import scrapy
+from scrapy.http import Request
+
+import pudl_scrapers
+
+
+class Eia860MSpider(scrapy.Spider):
+    """Spider for Monthly EIA 860."""
+
+    name = 'eia860m'
+    allowed_domains = ['www.eia.gov']
+
+    def __init__(self, year=None, month=None, *args, **kwargs):
+        """Spider for scrapping EIA 860 Monthly."""
+        super().__init__(*args, **kwargs)
+
+        if year is not None:
+            year = int(year)
+
+            if year < 2005:
+                raise ValueError("Years before 2005 are not supported")
+
+        self.year = year
+        self.month = month
+
+    def start_requests(self):
+        """Finalize setup and yield the initializing request."""
+        # Spider settings are not available during __init__, so finalizing here
+        settings_output_dir = Path(self.settings.get("OUTPUT_DIR"))
+        output_root = settings_output_dir / self.name
+        self.output_dir = pudl_scrapers.helpers.new_output_dir(output_root)
+
+        yield Request("https://www.eia.gov/electricity/data/eia860m/")
+
+    def parse(self, response):
+        """
+        Parse the eia860 home page.
+
+        Args:
+            response (scrapy.http.Response): Must contain the main page
+
+        Yields:
+            appropriate follow-up requests to collect XLSX files
+        """
+        if self.year is None:
+            yield from self.all_forms(response)
+
+        else:
+            yield self.form_for_month_year(response,
+                                           month=self.month,
+                                           year=self.year,)
+
+    # Parsers
+
+    def all_forms(self, response):
+        """
+        Produce requests for collectable EIA 860M forms.
+
+        Args:
+            response (scrapy.http.Response): Must contain the main page
+
+        Yields:
+            scrapy.http.Requests for Eia860M XLSX files from 2005 to the most
+            recent available
+        """
+        links = response.xpath(
+            "//table[@class='basic-table full-width']"
+            "//td[2]"
+            # "/a[contains(text(), 'XLS')]"
```

Comment:
What's up with this hanging around? Should it just be removed? Or should it be XLSX instead of XLS given the files available for `eia860m`?

Reply:
I... this was very hard for me to edit because I've never used xpath. It works fully without the `/a[contains]` but I added back in another restriction incase they ever add another table onto this page.